CONTRIBUTING.md

```diff
@@ -0,0 +1,68 @@
+# Contributing to the Self Hosted Show Wiki
+
+
+## Objectives of This Wiki
+
+In order to contribute well, you need to understand what the objectives are behind this wiki.
+
+What we are trying to create:
+- A compilation of tools, tips and tricks for self-hosting specific applications.
+- A place where beginners and experts can learn about how to quickly, efficiently and securely deploy applications on their own server(s).
+- Snippets that people can quickly reference in deploying applications.
+- A hub for quality documentation (to the extent possible).
+
+What we are not trying to do:
+- Be another large list of self-hosted tools. See [awesome-selfhosted](https://github.com/awesome-selfhosted/awesome-selfhosted) for a comprehensive list of self-hosted applications.
+- Be a hub for script kitties. People should _learn_ from documentation here not just come here to copy/paste. Snippets are important but should be documented properly.
+- Insecure, sloppy or any undesirable systems administration practices such as:
+  - Executing remote code by `curl`-ing an untrusted script and piping it to `sudo bash`: `curl https://example.com/awesomescript.sh | sudo bash`
+  - Installing additional bloat within a Docker container
+  - Etc.
+
+
+We want to encourage the use of dev/ops tools such as Docker, Ansible, SALT Stack, etc. for increased ease of use and for automating/standardizing as much as possible.
+
+
+## Directory Layout:
+
+The `docs` folder in the root of the repository is where wiki content should be placed.
+
+The following example directory layout should be adhered to when contributing:
+
+`docs/category/application/application.md`
+
+
+For example, if I wanted to add an article for Home Assistant, my layout could be this:
+
+`docs/home-automation/home-assistant/home-assistant.md`
+
+This would assume that the `home-assistant.md` entry belongs in the `home-automation` category. Look at the existing category directories before attempting to create a completely new category directory.
+
+
+Any images embedded in an article should be placed within a folder called `images` within the same directory:
+
+`docs/home-automation/home-assistant/images/example.png`
+
+
+Any other relevant resources (scripts, compose files, Ansible playbooks, etc.) pertaining to the application should be placed within the same folder the relevant application is using.
+
+
+## How to Contribute
```

Comment:
We also probably need a git basics primer here. I've already received a few emails from people not quite understanding forking / branching etc. We can probably link this to a sub-page in the wiki itself with a high level paragraph overview here.

Reply:
I could write something up either in the BUILD.md or something like setting_up_local_environment.md. The instructions would be like:
- Fork main project.
- Clone repository (your fork of the project) to your computer.
- Add upstream branch (Useful if needed to pull changes from upstream and merge them with own fork of it).
- Create and checkout into a "feature" branch.
- Add/commit changes (Writing meaningful commit messages).
- Push changes back to your fork on Github.
- Create Pull Request on Github.

Reply:
I removed BUILD.md altogether and placed the build instructions in CONTRIBUTING.md in spirit of https://github.com/selfhostedshow/wiki/pull/2#issuecomment-587112054